File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int , int>mp;
        int n = nums.size();
        for (int i = 0 ; i < n; i++)
        {
            int x = ++mp[nums[i]];
            if (x > 1)return 1;
        }
        return 0;
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Map<Integer, Boolean> mp = new HashMap<>();
        for (int num : nums)
        {
            if (mp.containsKey(num))return true;
            mp.put(num , true);
        }
        return false;
    }
}